packages/core/src/router.ts

```diff
@@ -39,3 +39,29 @@ export function newRouterExecuteTransactionRequest(
 
   return { to: getContractAddress(chainId, 'Router'), data, value };
 }
+
+export interface NewRouterExecuteWithSignerFeeTransactionOptions {
+  chainId: number;
+  routerBatchLogics: IParam.LogicBatchStruct;
+  signer: string;
+  signature: BytesLike;
+  tokensReturn?: string[];
+  value?: BigNumberish;
+  referralCode?: number;
+}
+
+export function newRouterExecuteWithSignerFeeTransactionRequest(
+  options: NewRouterExecuteWithSignerFeeTransactionOptions
+): common.TransactionRequest {
+  const { chainId, routerBatchLogics, signer, signature, tokensReturn = [], value = 0, referralCode = 0 } = options;
```

Comment:
Why would we set `0` for `value` and `referralCode`? If that's the case, do users still need to provide these 2 params?

Reply:
It should be the way to assign default value since they are optional.